File: types/survey.ts
export interface Survey {
  id: string
  creator_id: string
  title: string
  description: string
  category: string
  reward_amount: number
  estimated_duration: number
  target_responses: number
  current_responses: number
  status: 'draft' | 'pending' | 'approved' | 'rejected' | 'active' | 'completed'
  created_at: string
  updated_at: string
  expires_at?: string
  questions?: Question[]
  // Backward compatibility aliases
  reward?: number // alias for reward_amount
  responses_count?: number // alias for current_responses
  responses?: number // alias for current_responses
  target?: number // alias for target_responses
  expiresAt?: string // alias for expires_at
  eligible?: boolean
}

export interface CreatorSurvey extends Survey {
  // Ensure backward compatibility fields are present
  responses: number
  target: number
  expiresAt: string
}

export interface Question {
  id: string
  survey_id: string
  type: 'single' | 'multi' | 'text' | 'rating' | 'matrix' | 'multiple_choice' | 'open_ended'
  text: string
  options?: string[]
  required?: boolean
  order_num: number
  placeholder?: string
  scale?: number
  rows?: string[]
  cols?: string[]
}

export interface SurveyResponse {
  question_id: string
  answer: string
}

File: backend/models/survey.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Survey struct {
	ID                uuid.UUID  `json:"id" db:"id"`
	CreatorID         uuid.UUID  `json:"creator_id" db:"creator_id"`
	Title             string     `json:"title" db:"title"`
	Description       string     `json:"description" db:"description"`
	Category          *string    `json:"category" db:"category"`
	RewardAmount      int        `json:"reward_amount" db:"reward_amount"`
	EstimatedDuration int        `json:"estimated_duration" db:"estimated_duration"`
	TargetResponses   int        `json:"target_responses" db:"target_responses"`
	CurrentResponses  int        `json:"current_responses" db:"current_responses"`
	Status            string     `json:"status" db:"status"`
	CreatedAt         time.Time  `json:"created_at" db:"created_at"`
	UpdatedAt         time.Time  `json:"updated_at" db:"updated_at"`
	ExpiresAt         *time.Time `json:"expires_at" db:"expires_at"`
}
